timeout.go

```diff
@@ -0,0 +1,35 @@
+package codacytool
+
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+func timeoutSeconds() time.Duration {
+	value, exists := os.LookupEnv("TIMEOUT_SECONDS")
+
+	if !exists {
+		return defaultTimeout
+	}
+
+	seconds, err := strconv.Atoi(value)
+	if err != nil {
+		return defaultTimeout
+	}
+
+	return time.Duration(seconds) * time.Second
+}
+
+func runMethodWithTimeout(method func(), timeoutExceeded func(), maxDuration time.Duration) {
```

Comment:
it is my understanding that the "production" way of doing this is using:
https://golang.org/pkg/context/#WithTimeout

Reply:
Nice suggestion! 👌
Added `context` package to add the timeout feature.